### PR TITLE
feat(Webhook Node): Overhaul

### PR DESCRIPTION
**Pull request by michael-radency**
## Summary
Overhaul

## Related tickets and issues
https://linear.app/n8n/issue/NODE-1229/webhook-node-overhaul

### CODE REVIEW

File: packages/nodes-base/nodes/Webhook/utils.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import { NodeConnectionType, NodeOperationError } from 'n8n-workflow';
+import type { IDataObject, IWebhookFunctions, INodeExecutionData } from 'n8n-workflow';
+
+export const getResponseCode = (parameters: IDataObject) => {
```
Comment (elsmr) [inline packages/nodes-base/nodes/Webhook/utils.ts:4]: Would be nice to have an interface for the parameters instead of `IDataObject`

**Review by elsmr — COMMENTED** [review]
Works great 🎉 left some minor comments/questions

File: packages/nodes-base/nodes/Webhook/utils.ts [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
+	options?: {
+		responseCode?: number;
+		noResponseBody?: boolean;
+	};
+};
+
+export const getResponseCode = (parameters: WebhookParameters) => {
 	if (parameters.responseCode) {
 		return parameters.responseCode;
 	}
```
Comment (elsmr) [inline packages/nodes-base/nodes/Webhook/utils.ts:18]: There is a typo below `resposeCode` -> `responseCode`

File: packages/nodes-base/nodes/Webhook/utils.ts [hunk 3]
```diff
@@ -1,7 +1,18 @@
...
 import type { IDataObject, IWebhookFunctions, INodeExecutionData } from 'n8n-workflow';
 
-export const getResponseCode = (parameters: IDataObject) => {
+type WebhookParameters = {
+	httpMethod: string;
+	responseMode: string;
+	responseData: string;
+	responseCode?: number; //typeVersion <= 1.1
+	options?: {
+		responseCode?: number;
```
Comment (elsmr) [inline packages/nodes-base/nodes/Webhook/utils.ts:10]: I think this is

```
number | {values: { responseCode: number } | { customCode: number } }
```

File: packages/nodes-base/nodes/Webhook/utils.ts (hunk 2 above)
Comment (elsmr) [reply packages/nodes-base/nodes/Webhook/utils.ts:18]: There's a lot of `as IDataObject` in this file that is no longer needed I think

**Review by elsmr — DISMISSED** [review]

**Review by elsmr — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.36.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.36.0)

**Pull request merged**